### PR TITLE
- added markdown documentation for ActionSingle.

### PR DESCRIPTION
**Pull request by fabiosassu-sysdata**
- added changelog for version 1.0.1.
- fixed abnormal behavior on ActionQueue.
- Added the possibility to add the action instance instead of the class. This is useful for dependency injection (i.e. Dagger).
- Added ActionSingle.
- fixed the behavior of the error post in the observeFailure function.
- Updated Gradle plugin version.

### CODE REVIEW

File: Action.md [hunk 1]
```diff
@@ -19,7 +19,24 @@ To execute a **ActionQueue** as for the **Action** we have to link it to an inte
 
 ![alt text](https://github.com/SysdataSpA/KTAndroidArchitecture/blob/usecase_documentation/action_queue_execute.png "ActionQueue")
 
-### 1.3 Sample
+### 1.3 What is an ActionSingle
```
Comment (aguitto) [inline Action.md]: Change the name to SingleAction

File: ktandroidarchitecturecore/src/main/java/it/sysdata/ktandroidarchitecturecore/interactor/Action.kt [hunk 2]
```diff
@@ -31,19 +33,35 @@ class Action<Params : ActionParams, Model : Any, UiModel : Any> private construc
...
+
     /**
      * Execute the action
      *
      * @param params for use case
      */
     fun execute(params: Params) {
-        loadingLiveData.postValue(true)
+        lastParams = params
+//        loadingLiveData.postValue(true)
```
Comment (aguitto) [inline ktandroidarchitecturecore/src/main/java/it/sysdata/ktandroidarchitecturecore/interactor/Action.kt]: remove these lines

File: ktandroidarchitecturecore/src/main/java/it/sysdata/ktandroidarchitecturecore/interactor/ActionSingle.kt [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+import it.sysdata.ktandroidarchitecturecore.functional.Either
+import it.sysdata.ktandroidarchitecturecore.platform.SingleLiveEvent
+
+/**
+ * [ActionSingle] allow to define [UseCase] and [MutableLiveData] for handle the success, loading and failure case
+ *
+ * of Use Case and it allow to set a function for mapping Model into UiModel
+ *
+ */
+class ActionSingle<Params : ActionParams, Model : Any, UiModel : Any> private constructor() {
```
Comment (aguitto) [inline ktandroidarchitecturecore/src/main/java/it/sysdata/ktandroidarchitecturecore/interactor/ActionSingle.kt]: rename to SingleAction

**Review by aguitto — CHANGES_REQUESTED** [review]

File: Action.md (hunk 1 above)
Comment (fabiosassu-sysdata) [reply Action.md]: name changed.

File: ktandroidarchitecturecore/src/main/java/it/sysdata/ktandroidarchitecturecore/interactor/Action.kt (hunk 2 above)
Comment (fabiosassu-sysdata) [reply ktandroidarchitecturecore/src/main/java/it/sysdata/ktandroidarchitecturecore/interactor/Action.kt]: removed.

File: ktandroidarchitecturecore/src/main/java/it/sysdata/ktandroidarchitecturecore/interactor/ActionSingle.kt (hunk 3 above)
Comment (fabiosassu-sysdata) [reply ktandroidarchitecturecore/src/main/java/it/sysdata/ktandroidarchitecturecore/interactor/ActionSingle.kt]: renamed to SingleAction.

**Pull request merged**